esp-hal/src/dma/mod.rs

```diff
@@ -2116,6 +2224,9 @@ pub trait RegisterAccess: crate::private::Sealed {
 pub trait RxRegisterAccess: RegisterAccess {
     #[cfg(gdma)]
     fn set_mem2mem_mode(&self, value: bool);
+
+    fn peripheral_interrupt(&self) -> Option<Interrupt>;
+    fn async_handler(&self) -> Option<InterruptHandler>;
```

Comment:
Long term I think these should be `#[cfg]`s rather than `Option`s, as it makes it easier to grok what's really happening.